File: v2/snippets/cpp/common/activation.cpp
#include <gtest/gtest.h>
#include "Javonet.h"
#include "ActivationCredentials.h"

using namespace JavonetNS::Cpp::Sdk;

namespace CppActivationTests {

	TEST(Integration, Test_Activation_WrongCredentials_Returns1) {
		remove("javonet.lic");
		try {
			Javonet::Activate(ActivationCredentials::yourEmail, "your-licence-key");
		}
		catch (std::exception& e) {
			EXPECT_EQ(std::string("Javonet activation result: -1. Javonet licence is invalid. \nMessage: ERROR:Activation key incorrect"), std::string(e.what()));
		}

	}

	TEST(Integration, Test_Activation_CorrectCredentials_Returns0) {
		remove("javonet.lic");
		// <Javonet_activate>
		auto result = Javonet::Activate(ActivationCredentials::yourEmail, ActivationCredentials::yourLicenseKey);
		// </Javonet_activate>
		EXPECT_EQ((int)0, result);

		// <Javonet_activate_without_credentials>
		auto result2 = Javonet::Activate("", "");
		// </Javonet_activate_without_credentials>
		EXPECT_EQ((int)0, result2);
	}

}

File: v2/snippets/java/common/activation.java
import com.javonet.sdk.java.Javonet;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

public class JvmActivationTests {

    @Test
    @Tag("integration")
    public void Test_Activation_CorrectCredentials_Returns0() {
        // <Javonet_activate>
        int result = Javonet.activate(ActivationCredentials.yourEmail, ActivationCredentials.yourLicenseKey);
        // </Javonet_activate>
        Assertions.assertEquals(0, result);

        // <Javonet_activate_without_credentials>
        int result2 = Javonet.activate("", "");
        // </Javonet_activate_without_credentials>
        Assertions.assertEquals(0, result2);
    }
}
